subprojects/docs/src/docs/release/notes.md

```diff
@@ -169,9 +117,75 @@ In addition, this behavior can be enabled from the command line for individual b
+
+**Note:** This is a _Gradle 5.0 feature preview_, which means it is a potentially breaking change that will be activated by default in Gradle 5.0. It can be turned on in Gradle 4.6+ by adding `enableFeaturePreview('IMPROVED_POM_SUPPORT')` in _settings.gradle_.
+
+### Compile/runtime scope separation in POM consumption
+
+Since Gradle 1.0, `runtime` scoped dependencies have been included in the Java compile classpath, which has some drawbacks:
+
+- The compile classpath is much larger than it needs to be, slowing down compilation.
+- The compile classpath includes `runtime` files that do not impact compilation, resulting in unnecessary re-compilation when these files change.
+
+Now, if this new behavior is turned on, the Java and Java Library plugins both honor the separation of compile and runtime scopes. Meaning that the compile classpath only includes `compile` scoped dependencies, while the runtime classpath adds the `runtime` scoped dependencies as well. This is in particular useful if you develop and publish Java libraries with Gradle where the api/implementation dependencies separation is reflected in the published scopes.
+
+**Note:** This is a _Gradle 5.0 feature preview_, which means it is a potentially breaking change that will be activated by default in Gradle 5.0. It can be turned on in Gradle 4.6+ by adding `enableFeaturePreview('IMPROVED_POM_SUPPORT')` in _settings.gradle_.
+
+### Dependency constraints for transitive dependencies
+
+With [dependency constraints](userguide/managing_transitive_dependencies.html#sec:dependency_constraints), Gradle adds a mechanism to express constraints over transitive dependencies which are used during dependency resolution. 
+
+    dependencies {
+        implementation 'org.apache.httpcomponents:httpclient'
+    }
+
+    dependencies {
+        constraints {
+            implementation('org.apache.httpcomponents:httpclient:4.5.3') {
+                because 'previous versions have a bug impacting this application'
+            }
+            implementation('commons-codec:commons-codec:1.11') {
+                because 'version 1.9 pulled from httpclient has bugs affecting this application'
+            }
+        }
+    }
+
+In the example, the version of `commons-codec` that is brought in transitively is `1.9`. With the constraint, we express that we need at least `1.11` and Gradle will now pick that version during dependency resolution.
+
+In the future, Gradle will also allow you to _publish_ dependency constraints when using the [Gradle module metadata format](https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/design/gradle-module-metadata-specification.md) (currently under development) such that library authors will be able to share these constraints with library consumers.
```

Comment:
I think we should omit this sentence entirely because it's not quite ready to announce. When we announce this, we need to have a thorough explanation about the design considerations taken into account, with working examples.

Reply:
Reworded, and removed the sentence below.

Reply:
I also moved the section up both in the preamble and here as all the improved POM support features work with constraints.